public/js/uploader.js

```diff
@@ -27,24 +27,42 @@ function parseFile(event) {
       })
 
       var table = $("<table/>").attr("id","preview");
-      // console.log(data);
+      var tableSize = findTableSize(data);
+
       for (var i in data) {
-          //console.log(data[i]);
           var tr="<tr>";
-          for (var key in data[i]) {
+          var key = 0;
+          while (tableSize > key) {
               //console.log(key);
-              tr+="<td>"+data[i][key]+"</td>";
+              if (typeof data[i][key] != 'undefined') {
+                tr+="<td>"+data[i][key]+"</td>";
+              } else {
+                tr+="<td>"+ "-" +"</td>";
+              }
+
+              key++;
           }
           tr+="</tr>";
           $("table").append(tr);
       }
       $("#loading").text("");
       // Backup uploaded data to Local Storage
-      localStorage.setItem("upload", data);
+      localStorage.setItem("upload", JSON.stringify(data));
     }
   });
 }
 
+function findTableSize(data) {
```

Comment:
Good idea, but it ended up not actually working. I'll come back to this in a later pull request - there's likely a better way to do it, but for the sake of having it working for the client meeting tomorrow, I'm gonna leave it.